### ISSUE TITLE
Sylius EntityRepository from resource bundle is not compatible with sonata admin

### ISSUE THREAD

**Issue by hanovruslan** [Sylius/Sylius#2614]
**Sylius EntityRepository from resource bundle is not compatible with sonata admin**

There is [Sylius Resource EntityRepository](https://github.com/Sylius/SyliusResourceBundle/blob/master/Doctrine/ORM/EntityRepository.php) find method

```
    public function find($id)
    {
        return $this
            ->getQueryBuilder()
            ->andWhere($this->getAlias().'.id = '.intval($id))
            ->getQuery()
            ->getOneOrNullResult()
        ;
    }
```

$id param is suppose to be scalar (int). And when project does not use sonata admin bundle (with [sonata-project/doctrine-orm-admin-bundle](https://github.com/sonata-project/SonataDoctrineORMAdminBundle) as ORM adapter) with sylius resource everything is fine. But when it does then [SonataDoctrineORMAdminBundle Model manager](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/master/Model/ModelManager.php) combining with SyliusResourceBundle ORM EntityRepository give an error

as you can see in
```
    /**
     * {@inheritdoc}
     */
    public function find($class, $id)
    {
        if (!isset($id)) {
            return null;
        }
        $values = array_combine($this->getIdentifierFieldNames($class), explode(self::ID_SEPARATOR, $id));
        return $this->getEntityManager($class)->getRepository($class)->find($values);
    }
```

$id as parameter of Sylius Resouce EntityRepository::find method will be non-scalar

**Fix**
Replace

```
    /**
     * @param mixed $id
     *
     * @return null|object
     */
    public function find($id)
    {
        return $this
            ->getQueryBuilder()
            ->andWhere($this->getAlias().'.id = '.intval($id))
            ->getQuery()
            ->getOneOrNullResult()
        ;
    }
```

with

```
    /**
     * @param mixed $id
     * @param int $lockMode
     * @param null $lockVersion
     *
     * @return null
     */
    public function find($id, $lockMode = LockMode::NONE, $lockVersion = null)
    {
        $result = null;
        try {
            $result = $this->_em->find($this->_entityName, $id, $lockMode, $lockVersion);
        } catch (ORMInvalidArgumentException $exception) {
            //do nothing 
        } catch (ORMException $exception) {
            //do nothing
        }

        return $result;
    }
```
Is PR required with this fix?

**Comment by stloyd** [comment]
This is not issue of Sylius, it's implementation differences. You look at manager, where Sylius gives you repository, those two are (at least should be) totally different things.

**Comment by hanovruslan** [comment]
PR would be nice! btw this is not a problem for app because it's repos can extend from any other base repo with custom find method 

@stloyd do you mean this is issue of sylius resource bundle?  Or non of sylius bundles at all ?

anyway, thanks for response!

**Comment by pjedrzejewski** [comment]
We will remove all that magic from repositories soon.

**Comment by michalmarcinkowski** [comment]
Will be fixed in #3854.
